googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,219 @@
+    job.setOutputKeyClass(LongWritable.class);
+    job.setOutputValueClass(Text.class);
+    context.setInput(Input.of(sourceConfig.referenceName,
+                              new SourceInputFormatProvider(JsonTextBigQueryInputFormat.class, conf)));
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Text> input, Emitter<StructuredRecord> emitter) throws IOException {
+    getOutputSchema();
+    StructuredRecord record = jsonTransform(input.getValue());
+    emitter.emit(record);
+  }
+
+  private StructuredRecord jsonTransform(Text jsonText) {
+    Map<String, String> map;
+    map = GSON.fromJson(jsonText.toString(), MAP_STRING_STRING_TYPE);
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field field : outputSchema.getFields()) {
+      String fieldName = field.getName();
+      Schema schema = field.getSchema();
+      if (schema.getType() == Schema.Type.INT) {
+        builder.set(fieldName, Integer.parseInt(map.get(fieldName)));
+      } else {
+        builder.set(fieldName, map.get(fieldName));
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * {@link PluginConfig} class for {@link BigQuerySource}
+   */
+  public static class BQSourceConfig extends PluginConfig {
+    private static final String IMPORT_QUERY = "importQuery";
+    private static final String PROJECT_ID = "projectId";
+    private static final String INPUT_TABLE_ID = "InputTableId";
+    private static final String JSON_FILE_PATH = "jsonFilePath";
+    private static final String TEMP_BUCKET = "tempBuketPath";
+    private static final String OUTPUT_SCHEMA = "outputSchema";
+    private static final String PROJECTID_DESC = "The ID of the project in Google Cloud";
+    private static final String TEMP_BUCKET_DESC = "the tempory google cloud storage directory to store the " +
+                                                   "intermediate result. Example: gs://bucketname/directoryname, " +
+                                                   "the directory should not be existed. User should delete this " +
+                                                   "directory afterward manually to avoid extra google storage charge.";
+    private static final String IMPORT_QUERY_DESC = "The SELECT query to use to import data from the specified table." +
+                                                    " example: SELECT TOP(corpus, 10) as title, COUNT(*) as " +
+                                                    "unique_words FROM [publicdata:samples.shakespeare], where " +
+                                                    "publicdata is the project name, smalples is the dataset name, " +
+                                                    "shakespare is the table name. This is optional, if empty, just " +
+                                                    "read the inputTable configured";
+    private static final String JSON_KEYFILE_DESC = "the credential json key file path";
```

Comment:
(I'm not sure about this description because I don't know what this is exactly.)
```
"JSON key file path for credentials";